app/views/user/books/index.html.erb

```diff
@@ -1,24 +1,26 @@
 <h3><%= current_user.first_name %>'s Book Shelf</h3>
 <%= link_to "Add Books to My Shelf", new_user_book_path %>
 
+<%= link_to "My Shipping Information", current_user.address.nil? ? new_address_path : user_account_path %>
```

Comment:
This conditional hides the `user_account_path` link until a user has made an address.
It solves the issue of clicking on the `Account Info` navbar link (which is moved here now) and getting an exception thrown for not having an address.